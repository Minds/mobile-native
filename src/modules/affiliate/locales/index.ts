import {
  useTranslation as useAppTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import i18n from 'utils/locales/';

// import en from './en';

const NAME_SPACE = 'AffiliateModule';

const en = {
  screenTitle: 'Affiliate Program',
  screenDescription:
    'You can share links with friends and audience members to purchase products on Minds through customized links and earn money on qualifying purchases.',
};

i18n.addResourceBundle('en', NAME_SPACE, en);

export const useTranslation = (options?: UseTranslationOptions) =>
  useAppTranslation(NAME_SPACE, options);
