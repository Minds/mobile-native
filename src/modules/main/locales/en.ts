/* eslint-disable prettier/prettier */
export type TranslationType = typeof en;

const en = {
  'Sign up': 'Sign up',
  Login: 'Login',
  'Reset account': 'Reset account',
  'Sign in': 'Sign in',
  'Go back': 'Go back',
  Next: 'Next',
};

export default en;
