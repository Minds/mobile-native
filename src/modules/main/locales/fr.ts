// import { TranslationType } from './en'
/**
 * Add TranslationType to the object
 * ex: const fr: TranslationType = ...
 */

const fr = {};

export default fr;
