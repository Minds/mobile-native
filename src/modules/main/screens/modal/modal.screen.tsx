import React from 'react';
import {
  Button,
  Keyboard,
  Pressable,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
import { RootStackParams } from '../../navigator';
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
  title: { textTransform: 'capitalize' },
});

export default function ModalScreen(): JSX.Element {
  const route = useRoute<RouteProp<RootStackParams, 'Modal'>>();

  const { message, title, secondAction, dismissAfter } = route.params;
  const navigation = useNavigation();
  const { t } = useTranslation('mainModule');

  React.useEffect(() => {
    if (dismissAfter) {
      // setTimeout(() => {
      //   navigation.goBack();
      // }, dismissAfter);
    }
  }, [dismissAfter, navigation]);

  return (
    <View style={[]}>
      <Pressable onPress={Keyboard.dismiss} style={[]}>
        <Text style={[styles.title]}>{title}</Text>
        <Text style={[]}>
          {t(
            'Something has gone wrong. Should this error continue please contact our helpdesk',
          )}
        </Text>
        <Text style={[]}>{`${t('Error')} - ${message}`}</Text>
        <View style={[]}>
          {!dismissAfter && (
            <Button title="" onPress={() => navigation.goBack()}>
              {t('Ok')}
            </Button>
          )}

          {secondAction && (
            <>
              <View style={[]} />
              <Button
                title=""
                onPress={() => {
                  secondAction();
                  navigation.goBack();
                }}>
                {t('Retry')}
              </Button>
            </>
          )}
        </View>
      </Pressable>
    </View>
  );
}

export { ModalScreen };
