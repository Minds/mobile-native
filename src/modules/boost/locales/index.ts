import {
  useTranslation as useAppTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import i18n from 'utils/locales';

import en from './en';

const NAME_SPACE = 'BoostModule';

export const useTranslation = (options?: UseTranslationOptions) =>
  useAppTranslation(NAME_SPACE, options);

export default i18n.addResourceBundle('en', NAME_SPACE, en);
