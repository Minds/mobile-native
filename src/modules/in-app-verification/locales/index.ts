import {
  useTranslation as useAppTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import i18n from 'utils/locales';
import 'services/commonLocale';

import en from './en';

const NAME_SPACE = 'InAppVerificationModule';

export const useTranslation = (options?: UseTranslationOptions) =>
  useAppTranslation(['common', NAME_SPACE], options);

export default i18n.addResourceBundle('en', NAME_SPACE, en);
