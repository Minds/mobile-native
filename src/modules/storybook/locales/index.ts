import {
  useTranslation as useAppTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import i18n from 'utils/locales';

import en from './en';
import es from './es';

const NAME_SPACE = 'storybookModule';

export const useTranslation = (options?: UseTranslationOptions) =>
  useAppTranslation(NAME_SPACE, options);

export default i18n
  .addResourceBundle('en', NAME_SPACE, en)
  .addResourceBundle('es', NAME_SPACE, es);
