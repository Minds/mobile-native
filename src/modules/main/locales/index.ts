import {
  useTranslation as useAppTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import i18n from 'utils/locales';

import en from './en';
import fr from './fr';

const NAME_SPACE = 'mainContainer';

export const useTranslation = (options?: UseTranslationOptions) =>
  useAppTranslation(NAME_SPACE, options);

export default i18n
  .addResourceBundle('en', NAME_SPACE, en)
  .addResourceBundle('fr', NAME_SPACE, fr);
