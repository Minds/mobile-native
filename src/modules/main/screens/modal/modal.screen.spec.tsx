import React from 'react';
import {
  render,
  cleanup,
  RenderAPI,
  waitFor,
  renderHook,
} from '@testing-library/react-native';
import { TFunction } from 'i18next';
import { ModalScreen } from './modal.screen';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { useRoute } from '@react-navigation/native';
import { useTranslation } from '../../locales';

describe('Modal page', () => {
  let component: RenderAPI;
  let t: TFunction<'mainModule', undefined>;
  beforeEach(async () => {
    t = renderHook(() => useTranslation()).result.current.t;
    (useRoute as jest.Mock).mockReturnValue({
      params: {
        title: t('title'),
        message: 'message',
        dismissAfter: 3,
      },
    });
    component = await waitFor(() =>
      render(
        <SafeAreaProvider
          initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
          <ModalScreen />
        </SafeAreaProvider>,
      ),
    );
  });

  afterEach(() => {
    cleanup();
  });

  it('Modal snapshot test', () => {
    expect(component.toJSON).toMatchSnapshot();
  });

  it('renders the modal title correctly ', async () => {
    const heading = await component.getByText('title');
    expect(heading).toBeTruthy();
  });
});
