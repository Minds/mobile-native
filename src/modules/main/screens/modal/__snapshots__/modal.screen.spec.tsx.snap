// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal page Modal snapshot test 1`] = `[Function]`;
